### PR TITLE
[INTERNAL] Update usages of CJS require/exports to ESM import/export

### PR DESCRIPTION
**Pull request by flovogt**
Update the code samples to use additionally ESM import and export statements. The respective code blocks are displayed in tabs "CommonJS" and "ESM".

Screenshot 2023-01-02 at 12 34 28
Screenshot 2023-01-02 at 12 34 35

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -82,19 +82,17 @@ This allows you to rely on UI5 Tooling for UI5-specific build functionality and
 All available APIs are documented in the [UI5 Tooling API Reference](https://sap.github.io/ui5-tooling/v3/api/index.html).
 
 ```js linenums="1"
-const {normalizer} = require("@ui5/project");
-const {builder} = require("@ui5/builder");
+import {graphFromPackageDependencies} from "@ui5/project/graph";
```
Comment (RandomByte) [inline docs/index.md:85]: I think it's fair to assume that many developers still work with CommonJS (just like we do with many of our internal scripts). Maybe we can put both flavors into respective [content tabs](https://squidfunk.github.io/mkdocs-material/reference/annotations/#in-content-tabs) (e.g. `ESM` and `CommonJS`)?

Same for the custom task/middleware examples.

Code looks fine 👍

**Review by RandomByte — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (flovogt) [reply docs/index.md:85]: Good idea! Will do that! And also provide a sample code in https://github.com/SAP/openui5-sample-app/tree/demo-server-middleware-extensibility-v3

Comment (flovogt) [reply docs/index.md]: Done

Comment (matz3) [reply docs/index.md]: But then we should still put **ESM** as the first tab, right?

Comment (flovogt) [reply docs/index.md]: Good point. I'll rearrange the samples

Comment (flovogt) [reply docs/index.md]: Done

File: docs/pages/TasksAndProcessors.md [hunk 2]
```diff
@@ -1,57 +1,213 @@
 # Tasks and Processors
```
Comment (matz3) [inline docs/pages/TasksAndProcessors.md:1]: I feel like I have asked this already in the past: Is there a reason why this page is not linked anywhere?

Comment (matz3) [reply docs/pages/TasksAndProcessors.md:1]: I also think that the examples we have in the existing documentation pages are sufficient and also better, so I think we can remove this page.

**Review by matz3 — APPROVED** [review]

File: docs/pages/TasksAndProcessors.md (hunk 2 above)
Comment (RandomByte) [reply docs/pages/TasksAndProcessors.md:1]: Yes, AFAIK we wanted to remove this page already in the past but linked to it from somewhere.

Comment (flovogt) [reply docs/pages/TasksAndProcessors.md:1]: Fine. I'll create a follow-up PR removing this page

Comment (flovogt) [reply docs/pages/TasksAndProcessors.md:1]: PR https://github.com/SAP/ui5-tooling/pull/742 covers this

**Pull request merged**